byzcoin/rostsimul.go

```diff
@@ -0,0 +1,231 @@
+	return nil, errors.New("not implemented")
+}
+
+// GetIndex always returns -1
+func (s *ROSTSimul) GetIndex() int {
+	return -1
+}
+
+// GetVersion returns the version stored,
+// which can be updated to test contracts.
+func (s *ROSTSimul) GetVersion() Version {
+	return s.Version
+}
+
+// GetNonce is not implemented.
+func (s *ROSTSimul) GetNonce() ([]byte, error) {
+	return nil, errors.New("not implemented")
+}
+
+// ForEach is not implemented.
+func (s *ROSTSimul) ForEach(func(k, v []byte) error) error {
+	return errors.New("not implemented")
+}
+
+// StoreAllToReplica stores all stateChanges, without checking for validity!
+func (s *ROSTSimul) StoreAllToReplica(scs StateChanges) (ReadOnlyStateTrie, error) {
+	for _, sc := range scs {
+		s.Values[string(sc.InstanceID)] = StateChangeBody{
+			StateAction: Update,
+			ContractID:  sc.ContractID,
+			Value:       sc.Value,
+			Version:     sc.Version,
+			DarcID:      sc.DarcID,
+		}
+	}
+	return s, nil
+}
+
+// GetSignerCounter is not implemented
+func (s *ROSTSimul) GetSignerCounter(id darc.Identity) (uint64, error) {
+	return 0, xerrors.Errorf("not yet implemented")
+}
+
+//
+// Convenience methods for easier usage in tests
+//
+
+// CreateBasicDarc stores a darc with the _sign and "invoke:darc.
+// evolve" rules set to the signer.
+func (s *ROSTSimul) CreateBasicDarc(signer *darc.Identity, desc string) (*darc.Darc,
+	error) {
+	if signer == nil {
+		id := darc.NewIdentityEd25519(cothority.Suite.Point())
+		signer = &id
+	}
+	ids := []darc.Identity{*signer}
+	d := darc.NewDarc(darc.InitRules(ids, ids), []byte(desc))
+	buf, err := d.ToProto()
+	if err != nil {
+		return nil, xerrors.Errorf("couldn't convert darc to protobuf: %+v",
+			err)
+	}
+	s.Values[string(d.GetBaseID())] = StateChangeBody{
+		Value:      buf,
+		Version:    0,
+		ContractID: ContractDarcID,
+		DarcID:     nil,
+	}
+	return d, nil
+}
+
+// CreateCoin stores a new coin under a new random ID.
+func (s *ROSTSimul) CreateCoin(name string, value uint64) (id InstanceID,
+	err error) {
+	coin := Coin{Name: NewInstanceID([]byte(name)),
+		Value: value}
+	id = NewInstanceID(random.Bits(256, true, random.New()))
+	coinBuf, err := protobuf.Encode(&coin)
+	if err != nil {
+		err = xerrors.Errorf("couldn't encode coin: %+v", err)
+	}
+	s.Values[string(id[:])] = StateChangeBody{
+		StateAction: Create,
+		ContractID:  coinID,
+		Value:       coinBuf,
+		Version:     0,
+		DarcID:      make([]byte, 32),
+	}
+	return
+}
+
+// CreateRandomInstance adds a new instance given by the contractID and the
+// contract-structure. The ID will be chosen randomly,
+// the darcID can be nil, in which case it will be initialized to all-zeros.
+func (s *ROSTSimul) CreateRandomInstance(cID string,
+	value interface{}, darcID darc.ID) (id InstanceID, err error) {
+	id = NewInstanceID(random.Bits(256, true, random.New()))
+	valueBuf, err := protobuf.Encode(value)
+	if err != nil {
+		err = xerrors.Errorf("couldn't encode coin: %+v", err)
```

Comment:
```suggestion
		err = xerrors.Errorf("couldn't encode value: %+v", err)
```

btw, `CreateCoin`, `CreateRandomInstance` & `SetCoin` have a similar postamble, can you use a common function?